src/store/traction/pacbio/libraries/actions.js

```diff
@@ -45,32 +45,36 @@ const deleteLibraries = async ({ getters }, libraryIds) => {
   return responses
 }
 
-const setLibraries = async ({ commit, getters }) => {
+const setLibraries = async ({ commit, getters }, filter) => {
   let request = getters.libraryRequest
-  let promise = request.get({ include: 'request,tag,tube' })
-  let response = await handlePromise(promise)
-  let libraries = null
-
-  if (response.successful && !response.empty) {
-    // TODO: this is a hack. We are no longer returning multiple tags
-    // for a library so we should just have a single group_id
-    libraries = response.deserialize.libraries.map((library) => {
-      // This is getting more complicated
-      // I think this needs to be done in one go when we deserialize
-      // the libraries
-      const {
-        tag: { group_id: tag_group_id } = { group_id: null },
-        // should be request: { sample: { name } }
-        request: { sample_name } = { sample_name: null },
-        tube: { barcode } = { barcode: null },
-      } = library
-      return { ...library, tag_group_id, sample_name, barcode }
-    })
+  let promise = request.get({
+    include: 'request,tag,tube,pool',
+    filter: filter,
+  })
+  let response = await handleResponse(promise)
+
+  const { success, data: { data, included = [] } = {}, errors = [] } = response
+  const { tubes, tags, requests, pools } = groupIncludedByResource(included)
 
+  if (success) {
```

Comment:
lots going on here. Maybe a bit more explanation. Do we do anything similar elsewhere?

Reply:
Couldn't find a similar method but added a small explanation.